nixos/modules/services/networking/ntp/ntpd-rs.nix

```diff
@@ -0,0 +1,89 @@
+{ lib, config, pkgs, ... }:
+
+let
+  cfg = config.services.ntpd-rs;
+  format = pkgs.formats.toml { };
+  configFile = format.generate "ntpd-rs.toml" cfg.settings;
+in
+{
+  options.services.ntpd-rs = {
+    enable = lib.mkEnableOption "Network Time Service (ntpd-rs)";
+    metrics.enable = lib.mkEnableOption "ntpd-rs Prometheus Metrics Exporter";
+
+    package = lib.mkPackageOption pkgs "ntpd-rs" { };
+
+    useNetworkingTimeServers = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Use source time servers from {var}`networking.timeServers` in config.
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = format.type;
+      };
+      default = { };
+      description = lib.mdDoc ''
+        Settings to write to {file}`ntp.toml`
+
+        See <https://docs.ntpd-rs.pendulum-project.org/man/ntp.toml.5>
+        for more information about available options.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !config.services.timesyncd.enable;
```

Comment:
shouldn't ntp.service chrony.service
also get an assertion for conflict ?
just looking at the systemd service file, it seems that they would be problematic too
https://github.com/pendulum-project/ntpd-rs/blob/main/docs/examples/conf/ntpd-rs.service
(happy to make a PR for it myself, just double checking).

Reply:
This is a general problem with all NTP implementations (like ntpd or openntpd). I've kept it out of this PR to not delay merging this module any further. We only check for conflicts with the default NTP server in NixOS, which is systemd-timesyncd. In other cases the user should know what she's doing. Someone needs to comes up with a proper solution.